library/Notifications/Hook/ObjectsRendererHook.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/* Icinga Notifications Web | (c) 2024 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Notifications\Hook;
+
+use Exception;
+use Icinga\Application\Hook;
+use Icinga\Application\Logger;
+use Icinga\Module\Notifications\Model\Objects;
+use ipl\Html\HtmlString;
+use ipl\Html\ValidHtml;
+use ipl\Web\Url;
+use ipl\Web\Widget\Link;
+
+abstract class ObjectsRendererHook
+{
+    /**
+     * Array of Object ID tags for each source
+     *
+     * It has the following structure : ['object source type' => ['object ID' => [object ID tags]]]
+     *
+     * @var array<string, array<string, array<string, string>>>
+     */
+    public static $objectIdTagsCache = [];
+
+    /**
+     * Array of HTMLs for object names with their corresponding object IDs as keys
+     *
+     * It has the following structure : ['object ID' => registered HTML for the object name]
+     *
+     * @var array<string, ValidHtml>
+     */
+    protected static $registeredObjectNameHtmls = [];
+
+    /**
+     * Get the HTML for the object names for the objects using the object ID tags
+     *
+     * @param array<string, array<string, string>> $objectIdTags Array of object ID tags with their corresponding
+     *                                                           object IDs as keys. It has the following structure:
+     *                                                           [''object ID' => [object ID tags]]
+     *
+     * @return array<string, ValidHtml> Array of HTMLs for object names with their object IDs as keys
+     */
+    abstract public function getHtmlForObjectNames(array $objectIdTags): array;
+
+    /**
+     * Get the source type of the objects
+     *
+     * @return string
+     */
+    abstract public function getSourceType(): string;
+
+    /**
+     * Create the object link for the given object ID tag
+     *
+     * @param array<string, string> $objectIdTag
+     *
+     * @return ValidHtml
+     */
+    abstract public function createObjectLink(array $objectIdTag): ValidHtml;
+
+    /**
+     * Register HTMLs to be rendered for the object names using the cached object ID tags
+     *
+     * @return void
+     */
+    final public static function registerHtmlForObjectNames(): void
```

Comment:
Please wrap the function body in a `new private static method` and call it here.  his new method can then also be called in `getObjectName` if this registration does not take place. For example, if the object is not a list item.

Function name : e.g. `prepare(Array)`

Reply:
If an HTML for the object name is not already loaded, I don't think doing this has any effect. If you are suggesting this because of Event/Incident detail view shown in a single column. It would still be unnecessary as the object is registered and the HTMLs for the object is already prepared. If not then it should be prepared from the object ID tags.

Reply:
After discussing it offline. I think, I understand this suggestion better.